File: Logitude/Logitude.BL/CommonDataModel/CustomFilters/UserCustomFilter.cs
using System;
using System.Linq;
using System.Collections.Generic;
using Simplog.Data.CommonDataModel.EntityPOCOs;
using Simplog.Server.Infrastructure.DataContracts;

namespace Logitude.BL.CommonDataModel.CustomFilters
{
    public class UserCustomFilter
    {
        private int tenant;
        public UserCustomFilter(int tenant)
        {
            this.tenant = tenant;
        }

        public IQueryable<User> GetFilteredQuery(QueryOperations operations, IQueryable<User> iQueryableData)
        {
            List<QueryFilterItem> queryFilters = operations.QueryFilterItems;

            foreach (QueryFilterItem item in queryFilters)
            {
                if (item.IsCustom)
                {
                    if (item.FieldName == "ActiveUsers")
                    {
                        iQueryableData = iQueryableData.Where(d => d.Contact.InActive == false);
                    }

                    if (item.FieldName == "InactiveUsers")
                    {
                        iQueryableData = iQueryableData.Where(d => d.Contact.InActive == true);
                    }

                    if (item.FieldName == "ActiveLicensed")
                    {
                        iQueryableData = iQueryableData.Where(d => d.LicencedUser == true && d.Contact.InActive == false);
                    }

                    if (item.FieldName == "ActiveNotLicensed")
                    {
                        iQueryableData = iQueryableData.Where(d => d.LicencedUser == false && d.Contact.InActive == false);
                    }

                    if (item.FieldName == "EmployeeGroupCustomFilter")
                    {
                        //var groupId = item.FieldValue.ToString();
                        //if (!string.IsNullOrEmpty(groupId))
                        //{
                        //    iQueryableData = iQueryableData.Where(d => d.GroupId != null && d.GroupId.Any(a => a == groupId));

                        //}
                    }

                    if (item.FieldName == "HasEmail")
                    {
                        iQueryableData = iQueryableData.Where(c => !string.IsNullOrEmpty(c.Contact.Email));
                    }

                    if (item.FieldName == "SearchEmailsWithout")
                    {
                        string value = item.FieldValue as string;
                        if (!string.IsNullOrEmpty(value))
                        {
                            string[] emails = value.Split(';');
                            if (emails.Count() > 0)
                            {
                                iQueryableData = iQueryableData.Where(c => !emails.Contains(c.Contact.Email));
                            }
                        }
                    }
                }
            }

            return iQueryableData;
        }
    }
}

File: Logitude/AngularModules/AngularModules/Common/CustomFilters/UserCustomFilter.ts
import {UserList} from '../EntityLists/UserList';
import {AppTool} from '../../Infrastructure/Tools';

export class UserCustomFilter {

    public static  GetFilteredQuery(addtionalFiltersValues: any, data: UserList[]) {

        //addtionalFiltersValues = addtionalFiltersValues.replace('[', '').replace(']','');
        var mykeys = JSON.parse(addtionalFiltersValues);
        for (var i in mykeys) {
            var propName = mykeys[i];
            if (propName.FieldName == "EmployeeGroupCustomFilter") {
                var groupId = propName.FieldValue;
                if (!AppTool.IsNullOrEmpty(groupId)) {
                    data = data.filter(d => d.GroupId != null && (d.GroupId.indexOf(groupId) > -1));//.indexOf(groupId) > -1);
                }
            }
        }

        return data;
        //for (var i = 0; i < addtionalFiltersValues.length; i++) {
        //    var pair = addtionalFiltersValues[i].split('=');
        //    if (decodeURIComponent(pair[0]) == "AdditionalFilters") {
        //        var obj = JSON.parse(pair[1]);
        //        for (var key in obj) {
        //            if (key == 'EmployeeGroupCustomFilter') {
        //                var groupId = obj[key];
        //                if (!AppTool.IsNullOrEmpty(groupId)) {

        //                    data = data.filter(d => d.GroupId != null && d.GroupId.indexOf(groupId) > -1);
        //                }

        //                return data;
        //            }
        //        }
        //    }
        //}
    }
}